pkg/build/api/validation/validation_test.go

```diff
@@ -116,8 +150,19 @@ func TestBuildConfigValidationFailure(t *testing.T) {
 			},
 		},
 	}
-	if result := ValidateBuildConfig(buildConfig); len(result) != 2 {
-		t.Errorf("Unexpected validation result %v", result)
+	errors := ValidateBuildConfig(buildConfig)
+	if len(errors) != 1 {
+		t.Fatalf("Unexpected validation errors %v", errors)
+	}
+	err := errors[0].(*fielderrors.ValidationError)
+	if err.Type != fielderrors.ValidationErrorTypeInvalid {
+		t.Errorf("Unexpected error type, expected %s, got %s", fielderrors.ValidationErrorTypeInvalid, err.Type)
+	}
+	if err.Field != "triggers" {
+		t.Errorf("Unexpected field name expected name, got %s", err.Field)
+	}
+	if !strings.Contains(err.Detail, "only one ImageChangeTrigger is allowed") {
+		t.Errorf("Unexpected error details: %s", err.Detail)
```

Comment:
It couples that's true, but I want to make sure about the error we are throwing, but I like the idea of constants. On the other hand would we create so many to cover every single use case. Whereas testing details message is not always done actually, only certain cases which is ok imho.